tools/build/ms-plugins.coffee

```diff
@@ -0,0 +1,113 @@
+{ walkFiles } = require('@resin.io/doxx-utils')
+HbHelper = require('@resin.io/doxx-handlebars-helper')
+
+path = require('path')
+fse = require('fs-extra')
+semver = require('semver')
+
+utils = require('../lib/utils')
+pythonHelper = require('../lib/python')
+
+
+removeFile = (obj, file, files) ->
+	if obj and obj is utils.UNSUPPORTED
+		delete files[file]
+
+copyFile = (src, dest) ->
+	try
+		fse.copySync(src, dest)
+	catch err
+		console.error(err)
+
+copyDebianFile = (obj, destDir) ->
```

Comment:
this may be the best solution at hand, but to me it feels really weird that we do some fs manipulations from the MS plugin. of course, you can do that, but it breaks the abstraction. MS plugins are supposed to manipulate the abstract files tree
instead of this one can use the existing mechanisms: put these files in the tree and annotate them properly to be orchestrated by MS. Like take the file `./supporting_files/debian.01_buildconfig`, place it like `src/debian.01_buildconfig` and add this front-matter

```
---
dynamic:
  variables: [ $debian_arch, $debian_suite ]
  ref: debian/$debian_arch/$debian_suite/01_buildconfig
  skip_ext: true
---
```

Same for other files. For files that are for example wheezy-only it can be

```
---
dynamic:
  variables: [ $debian_arch ]
  ref: debian/$debian_arch/wheezy/entry.sh
---
```

I see how there's a problem with "use this entry.sh for all _except of_ wheezy". This can be solved by having a placeholder file that imports a single partial `{{ import "debian-entry" }}`. That entry will have a `_default.part`, and then a single overload for `wheezy.part`.

Finally for files that need to be present for every suite except of wheezy (and be missing completely for wheezy) you can use the `UNSUPPORTED` trick placing this keyword into the partial itself (and then killing the files that contain this word in one final pass).